src/renderers/shared/ReactDebugTool.js

```diff
@@ -110,14 +112,15 @@ function beginLifeCycleTimer(debugID, timerType) {
     return;
   }
   warning(
-    !currentTimerType,
+    !currentTimerType || lifeCycleTimerHasWarned,
     'There is an internal error in the React performance measurement code. ' +
     'Did not expect %s timer to start while %s timer is still in ' +
     'progress for %s instance.',
     timerType,
     currentTimerType || 'no',
     (debugID === currentTimerDebugID) ? 'the same' : 'another'
   );
+  lifeCycleTimerHasWarned = true;
```

Comment:
This sets it regardless of whether the warning has occurred. Sorry I didn't realize this at first.

It seems like it would indeed be easier to add an `if (currentTimetType)` here. The warning would just get `lifeCycleTimerHasWarned` as an argument. 

Same below. 

Sorry I didn't see this at first!